File: Client/Captions/CaptionsLib/source/encore/EncoreUtils.vb
'Filename: EncoreUtils.vb
'Version: 20121015

Imports ClipFlair.CaptionsLib.utils.DateTimeUtils

Namespace ClipFlair.CaptionsLib.Encore

  Public NotInheritable Class EncoreUtils
    'for Adobe Encore

    Public Const EncoreTimeFormat As String = "HH:mm:ss:ff"
    Public Const SignificantDigits As Integer = 2

    Public Shared BaseTime As DateTime = DATETIMEZERO

    Public Shared Function SecondsToEncoreTime(ByVal seconds As Double) As String
      Return SecondsToDateTimeStr(seconds, BaseTime, EncoreTimeFormat, SignificantDigits)
    End Function

    Public Shared Function EncoreTimeToSeconds(ByVal EncoreTime As String) As Double
      Return TimeStrToSeconds(EncoreTime, BaseTime, EncoreTimeFormat, SignificantDigits)
    End Function

  End Class

End Namespace

File: Client/Captions/CaptionsLib/source/encore/EncoreUtils.cs
//Filename: EncoreUtils.cs
//Version: 20131105

using System;
using ClipFlair.CaptionsLib.Utils;

namespace ClipFlair.CaptionsLib.Encore
{

  public sealed class EncoreUtils
  {
    //for Adobe Encore

    public const string EncoreTimeFormat = "HH:mm:ss:ff";

    public const int SignificantDigits = 2;

    public static DateTime BaseTime = DateTimeUtils.DATETIMEZERO;
    public static string SecondsToEncoreTime(double seconds)
    {
      return DateTimeUtils.SecondsToDateTimeStr(seconds, BaseTime, EncoreTimeFormat, SignificantDigits);
    }

    public static double EncoreTimeToSeconds(string EncoreTime)
    {
      return DateTimeUtils.TimeStrToSeconds(EncoreTime, BaseTime, EncoreTimeFormat, SignificantDigits);
    }

  }

}